### PR TITLE
Feature/calendar screen rebuild

### PR DESCRIPTION
**Pull request by mateki0**

### CODE REVIEW

File: src/screens/calendar/Calendar.tsx [hunk 1]
```diff
@@ -1,95 +1,411 @@
...
+      ) : (
+        <Box
+          backgroundColor="lightGrey"
+          flex={1}
+          borderRadius="l"
+          marginTop={singlePreviousEvent ? 'xxxl' : 'l'}
+          paddingTop="xxxl">
+          <AnimatedBox flex={1} style={animatedStyles}>
+            <Box paddingBottom="xm" justifyContent="center" alignItems="center">
+              <PanGestureHandler onGestureEvent={panGestureHandler}>
```
Comment (lukbar48) [inline src/screens/calendar/Calendar.tsx]: can it be replaced with GestureRecognizer?

File: src/screens/calendar/Calendar.tsx [hunk 2]
```diff
@@ -1,95 +1,411 @@
...
+
+  useEffect(() => {
+    if (slicedRequests.length > 0) {
+      scrollToIndex(0)
+    }
+  }, [slicedRequests.length])
+
+  const disableSetDateButton = periodStart?.length < 9 && periodEnd?.length < 9
+
+  const springConfig = {
```
Comment (lukbar48) [inline src/screens/calendar/Calendar.tsx]: this can be outside component

File: src/screens/calendar/Calendar.tsx [hunk 3]
```diff
@@ -1,95 +1,411 @@
...
+
+  const clearDatesInputs = () => {
+    handleSetPeriodStart('')
+    handleSetPeriodEnd('')
+    setSlicedRequest([])
+    handleSetPreviousEvent()
+  }
+
+  useEffect(() => {
+    if (slicedRequests.length > 0) {
```
Comment (lukbar48) [inline src/screens/calendar/Calendar.tsx]: I think in most of if/elses braces in this component can be removed, especially when we have so much code here

File: src/screens/calendar/Calendar.tsx [hunk 4]
```diff
@@ -1,95 +1,411 @@
...
+      if (singlePreviousEvent?.date !== prevEvent?.date) {
+        setSinglePreviousEvent(prevEvent)
+      }
+    },
+    [isPeriodStartLowerThanRequestsDate, periodStart, requestsDays, singlePreviousEvent?.date]
+  )
+
+  useEffect(() => {
+    handleSetPreviousEvent()
+    // we want to set this only on initial render
```
Comment (lukbar48) [inline src/screens/calendar/Calendar.tsx]: do we want to write this when dependency array says this?

File: src/screens/calendar/Calendar.tsx [hunk 5]
```diff
@@ -1,95 +1,411 @@
...
+const today = date.toISOString().split('T')[0]
+
+const isEndDateLowerThanStartDate = (startDate: string, endDate: string) => {
+  const startTimestamp = new Date(startDate).getTime()
+  const endTimestamp = new Date(endDate).getTime()
+
+  return endTimestamp < startTimestamp
+}
+
+const AnimatedBox = Animated.createAnimatedComponent(Box)
```
Comment (lukbar48) [inline src/screens/calendar/Calendar.tsx]: you can import this from AnimatedBox.tsx

File: src/screens/calendar/Calendar.tsx [hunk 6]
```diff
@@ -1,95 +1,411 @@
...
+      notify('infoCustom', notificationConfig)
+    } else {
+      handleSetEventsInPeriod(startIndex)
+    }
+
+    setWasDateChangePressed()
+    handleSetPreviousEvent()
+  }
+
+  const { navigate } = useNavigation<CalendarNavigatorType<'CALENDAR'>>()
```
Comment (lukbar48) [inline src/screens/calendar/Calendar.tsx]: i think this can stay at very top, because its used only used once in component at the bottom

**Review by lukbar48 — APPROVED** [review]

File: src/screens/calendar/Calendar.tsx (hunk 4 above)
Comment (mateki0) [reply src/screens/calendar/Calendar.tsx]: Even if someone will know how it behave, he may not know why I've disabled it to work like that. I've changed the comment message

File: src/screens/calendar/Calendar.tsx (hunk 1 above)
Comment (mateki0) [reply src/screens/calendar/Calendar.tsx]: I don't want to use it here. Let stay with using GestureRecognizer for screen's wrapping like it is done now :)

**Pull request merged**